webapp/src/components/Staking/LiquidityGaugeModal/StakingActionModal.tsx

```diff
@@ -147,13 +185,30 @@ const StakingActionModal: React.FC<StakingActionModalProps> = ({
       assetToFiat(poolSizeInAsset, prices[asset], decimals)
     );
 
-    return [
-      `${
-        poolSizeInUSD > 0
-          ? (((1 + poolRewardInUSD / poolSizeInUSD) ** 52 - 1) * 100).toFixed(2)
-          : "0.00"
-      }%`,
-    ];
+    const baseRewards =
+      poolSizeInUSD > 0
+        ? ((1 + poolRewardInUSD / poolSizeInUSD) ** 52 - 1) * 100
+        : 0;
+    const boostedMultiplier = calculateBoostMultiplier({
+      workingBalance: lg5Data.workingBalances,
+      workingSupply: lg5Data.workingSupply,
+      gaugeBalance: parseUnits(input || "0", decimals),
+      poolLiquidity: lg5Data.poolSize,
+      veRBNAmount: votingPower,
+      totalVeRBN: totalVeRBN || BigNumber.from("0"),
+    });
+
+    const boostedRewards =
+      boostedMultiplier > 0 ? baseRewards * boostedMultiplier - baseRewards : 0;
+
+    return {
+      totalApy: `${(baseRewards + boostedRewards).toFixed(2)}%`,
+      baseRewards: `${baseRewards.toFixed(2)}%`,
+      boostedMultiplier: boostedMultiplier
+        ? `(${boostedMultiplier.toFixed(2)}X)`
+        : "",
+      boostedRewards: `${boostedRewards.toFixed(2)}%`,
```

Comment:
It probably makes sense to abstract the `totalApy` and the `baseReward` % to math.ts so we can unit test it. Seems straightforward now but I foresee that it will get more hairy

Reply:
make sense